src/lib/sensor_sampler/sensorSampler.h

```diff
@@ -1,38 +1,30 @@
 #pragma once
 
-#include "stm32_rtc.h"
+#include "ll.h"
+#include "FreeRTOS.h"
+#include "timers.h"
 #include <stdint.h>
 #include <stdbool.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-typedef bool (*sensorSampleFn)();
-typedef bool (*sensorInitFn)();
-typedef bool (*sensorCheckFn)();
+// We're using a 32-bit variable for sample flags (with two reserved)
+#define MAX_SENSORS 30
 
-typedef struct {
-  uint64_t uptime;
-  RTCTimeAndDate_t rtcTime;
-  uint16_t address;
-  float voltage;
-  float current;
-} __attribute__((packed)) powerSample_t;
-
-typedef struct {
-  uint64_t uptime;
-  RTCTimeAndDate_t rtcTime;
-  float temperature;
-  float humidity;
-} __attribute__((packed)) humTempSample_t;
+/// Used to initialize sensor node
+#define SENSOR_NODE_INIT(init, sample, check) \
+    { .list = { \
+          .sensor = { .initFn = init, .sampleFn = sample, .checkFn = check }, \
+          .name = NULL, \
+          .timer = {0}, \
```

Comment:
```suggestion
          .timer = NULL, \
```

Reply:
This change fixed the test build for me. Took me a while to wrap my head around where this came from. All the comparisons for timer in the code are whether `timer == NULL`, and the original initialization prior to this PR was in `sensorSamplerAdd` where each item gets malloc'd and memset to 0.

Reply:
Ah! And just sharing something I just realized. I was wondering, since we were using `pvPortMalloc` for each sensor before, and since you removed that, where did the same info go and how is it allocated. Every place that `SENSOR_NODE_INIT` is called is in a static variable initializer in the individual sensor sampler cpp files, like htuSampler.cpp, loadCellSampler.cpp, powerSampler.cpp, etc. Those statically allocated items get passed to `sensorSamplerAdd` where we save a pointer to the node in `LLNodeAdd`.